### PR TITLE
Make clear that hooks are not supported in LiveComponents

### PR DESCRIPTION
**Pull request by maxmarcon**
I fell into the trap, and so have others. Maybe making it clear upfront would help

### CODE REVIEW

**Comment by rktjmp** [comment]
Do you have a reproduction of this failing? Given a basic `phx.new` with the standard modal (which is a live component) and the hook:

```js
const MyHook = {
  mounted(){
    console.log("Hook mounted", this)
    this.el.addEventListener("click", (event) => {console.log(event)})
  },
}

let csrfToken = document.querySelector("meta[name='csrf-token']").getAttribute("content")
let liveSocket = new LiveSocket("/live", Socket, {
  params: {_csrf_token: csrfToken},
  hooks: { MyHook }
})
````

```html
      <.simple_form
        for={@form}
        id="user-form"
        phx-target={@myself}
        phx-change="validate"
        phx-submit="save"
      >
        <div id="some-id" phx-hook="MyHook">a hook</div>
```

I get the mounted and click event console messages.

**Comment by maxmarcon** [comment]
See the note above the one I added 😉 This is about server-side hooks, not client-side.

Try to add a client-side hook to a LiveComponent. You'll end up [here](https://github.com/phoenixframework/phoenix_live_view/blob/main/lib/phoenix_live_view/lifecycle.ex#L89).

**Comment by rktjmp** [comment]
Ha! Sorry!!

File: lib/phoenix_live_view.ex [hunk 1]
```diff
@@ -1401,6 +1401,8 @@ defmodule Phoenix.LiveView do
   > Note: This function is for server-side lifecycle callbacks.
   > For client-side hooks, see the
   > [JS Interop guide](js-interop.html#client-hooks).
+    
+  > Note: Hooks are not supported in LiveComponents.
```
Comment (josevalim) [inline lib/phoenix_live_view.ex:1405]:
```suggestion
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by maxmarcon** [comment]
Ah cool! Thanks for the correction :)

**Pull request merged**